src/Plugins/ComplexCore/src/ComplexCore/Filters/AlignGeometries.cpp

```diff
@@ -0,0 +1,429 @@
+#include "AlignGeometries.hpp"
+
+#include "complex/DataStructure/Geometry/AbstractGeometry.hpp"
+#include "complex/DataStructure/Geometry/AbstractGeometry2D.hpp"
+#include "complex/DataStructure/Geometry/AbstractGeometry3D.hpp"
+#include "complex/DataStructure/Geometry/EdgeGeom.hpp"
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+#include "complex/DataStructure/Geometry/RectGridGeom.hpp"
+#include "complex/DataStructure/Geometry/VertexGeom.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/GeometrySelectionParameter.hpp"
+#include "complex/Utilities/DataArrayUtilities.hpp"
+
+#include <string>
+
+namespace
+{
+constexpr complex::int32 k_EMPTY_PARAMETER = -123;
+
+complex::FloatVec3 extractOrigin(const complex::AbstractGeometry* geometry)
+{
+  if(auto* image = dynamic_cast<const complex::ImageGeom*>(geometry))
```

Comment:
These `if` blocks could be replaced with a `switch` on `AbstractGeometry::Type`. And the same for the other functions.